development/rfcs/0010-postmortem-principles.md

```diff
@@ -0,0 +1,73 @@
+
+### We will create an environment of psychological safety.
+
+Witnessing the public shaming and blaming of a co-worker for mistakes is a surefire way of reducing the confidence and productivity of others. Future incidences will most likely be covered up and reinforce the blame cycle, leading to more incidents, making the system weaker overall.
+
+### Incidents are an opportunity to learn and reflect
+
+In the broadest sense, an incident is any unplanned work you encounter in your day-to-day operations. Good judgement is required to know when unplanned events require an incident response, as seemingly innocuous events may have terrifying root causes. This will come with experience running incident responses, which means it is better to err on running more than running less. Some examples of events that may trigger the incident process are:
+
+- User-visible downtime or degradation beyond a certain threshold
+- Data loss of any kind
+- On-call engineer intervention (release rollback, rerouting of traffic, etc.)
+- A resolution time above some threshold
+- A monitoring failure (which usually implies manual incident discovery)
+
+### We care about the details
+
+It is important to keep an up-to-date timeline of events during an incident as it will be the only moment in time when the information is fresh in your mind. Humans naturally collapse details as time goes by and important nuances will be lost. Similarly, it will give team members joining the incident later on a good overview of where the process is at.
+
+### Action items need ownership and accountability
+
+We can have the best postmortem process in the world, but have it amount to nothing unless we follow up on the action items identified and hold ourselves accountable to them. It is not enough to identify them, they must be put into backlogs and actioned with dedicated owners. Ideally, postmortems would be shared internally across the organisation (ex. community meetings, Show The Thing) and potentially publicly. We want to demonstrate that we take them and their outcomes seriously.
+
+### Time needs to be made for incidents and postmortems
+
+The creation, review and completion of postmortems should be given emphasis by the management team. It should be made clear that recording our past successes and failures is a priority. That there is time, and room, for a team to do their postmortem, do them correctly, and follow up as needed.
+
+## Drawbacks
+
+We assume that this is the best way for us to learn from past mistakes and successes. However, it requires significant investment of time and resources to do properly, which comes at the opportunity cost of other work. Should we not properly follow through on our commitments we may end up with a broken process that has wasted time and resources with no usable outcomes.
+
+## Alternatives
+
+Time invested into postmortems could be invested into other work, while incidents are centrally monitored and managed.
```

Comment:
I don't understand this alternative. Possible to expand?

One alternative (not ideal of course): only document incident with long downtime or damage to the users experience

Reply:
Thank you!